### ISSUE TITLE
False positive with svelte/no-navigation-without-resolve when importing typed path

### ISSUE THREAD

**Issue by thobson** [sveltejs/eslint-plugin-svelte#1370]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.36.0

### What version of `eslint-plugin-svelte` are you using?

3.12.3

### What did you do?

This works

```svelte
<script lang"ts>
  import { resolve } from '$app/paths';
</script>

<a href={resolve("/")}>home</a>
```

This also works

```svelte
<script lang="ts">
  import { resolve } from '$app/paths';
  const home = resolve("/");
</script>

<a href={home}>home</a>
```

However this generates a `Found a link with a url that isn't resolved` error:

```ts
// src/lib/routes
import { resolve } from '$app/paths';
export const home = resolve("/");
```

```svelte
<script lang="ts">
  import { home } from '$lib/routes';
</script>

<a href={home}>home</a>
```

### What did you expect to happen?

I would expect the externalised route to be treated the same as the one that is resolved inline.

### What actually happened?

`Found a link with a url that isn't resolved  svelte/no-navigation-without-resolve`

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/thobson/svelte-eslint-errors

See `src/routes/+page.svelte`

**Comment by marekdedic** [comment]
Hi, thanks fr the report, I believe this is the same issue as #1319, which I hope to fix soon.

**Comment by Blackwidow-sudo** [comment]
It also fails when your resolved paths are nested like:
```svelte
<script>
  const navItems = [
    { name: 'Home', url: resolve('/') },
    { name: 'About', url: resolve('/about') }, 
  ]
</script>

{#each navItems as item}
  <a href={item.url}>{item.name}</a>
{/each}
```

**Comment by tschortsch** [comment]
And when doing something like this:

```typescript
<script lang="ts">
  const resolveBlogUrl = (slug: string) => resolve('/blog/[slug]', { slug });
</script>

<a href={resolveBlogUrl('my-post')}>Link to My Post</a>
```

**Comment by marekdedic** [comment]
Hi, #1321 should fix the original issue.
@Blackwidow-sudo I opened a separate issue for the problem from your comment (and others): #1485
@tschortsch I opened a separate issue for the problem from your comment: #1486
